torch/lib/THPP/tensors/generic/THCTensor.cpp

```diff
@@ -6,14 +6,14 @@
   dynamic_cast<const THCTensor&>(tensor)
 #define const_storage_cast(storage) \
   dynamic_cast<const THCStorage<real>&>(storage)
-#define const_long_cast(tensor) \
-  dynamic_cast<const THCTensor<long>&>(tensor)
+#define const_longlong_cast(tensor) \
```

Comment:
This change is related to my comment above. Should we let `int64_t` be associated with `long` in here?